src/java/com/unifina/signalpath/utils/MessageChainUtil.java

```diff
@@ -59,17 +59,12 @@ public StreamMessage getStreamMessage(Stream stream, Date timestampAsDate, Map c
 		long timestamp = timestampAsDate.getTime();
 		long sequenceNumber = getNextSequenceNumber(key, timestamp);
 		String publisherId = getUser().getPublisherId();
-		MessageID msgId = new MessageID(stream.getId(), streamPartition, timestamp, sequenceNumber, publisherId);
+		MessageID msgId = new MessageID(stream.getId(), streamPartition, timestamp, sequenceNumber, publisherId, "");
```

Comment:
Do `this.msgChainId = IdGenerator.getShort();` in constructor and use it here. `MessageChainUtil` instance is (publisher, msgChainId)-specific. BTW wouldn't hurt to have some class-level comments about the purpose and functionality of this class.